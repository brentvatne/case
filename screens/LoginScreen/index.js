import React from 'react'
import {
  Image,
  StyleSheet,
  KeyboardAvoidingView,
} from 'react-native'

import { NavigationActions } from 'react-navigation'
import Store from '../../state/Store'
import { LoginWithData } from './components/LoginWithData'

import layout from '../../constants/Layout'

export default class LoginScreen extends React.Component {
  _resetStack() {
    NavigationActions.init()
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [
        NavigationActions.navigate({ routeName: 'main' }),
      ]
    })
    this.props.navigation.dispatch(resetAction)
  }

  render() {
    console.log('rendering login')
    return (
      <KeyboardAvoidingView behavior="padding" style={styles.container}>
        <Image
          style={styles.logo}
          source={require('../../assets/images/logo.png')}
        />
        <LoginWithData onLogin={this._resetStack.bind(this)} />
      </KeyboardAvoidingView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center', 
    justifyContent: 'center',
    padding: 20,
    backgroundColor: '#fff'
  },
  logo: {
    width: 100,
    height: 100,
  },
});