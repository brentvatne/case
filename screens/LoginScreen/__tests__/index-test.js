import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import LoginScreen from '../index'

test('renders correctly', () => {
  const tree = renderer.create(
    <LoginScreen />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
