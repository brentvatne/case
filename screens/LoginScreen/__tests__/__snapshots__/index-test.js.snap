// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    space={3}
    style={
      Array [
        Object {
          "marginVertical": 24,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1.5,
            "borderColor": "#ccc",
            "marginHorizontal": "12.5%",
            "marginVertical": 6,
            "paddingTop": 12,
            "width": "75%",
          },
          undefined,
        ]
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email address"
        style={
          Array [
            Object {
              "fontSize": 14,
              "height": 21,
              "paddingHorizontal": 6,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1.5,
            "borderColor": "#ccc",
            "marginHorizontal": "12.5%",
            "marginVertical": 6,
            "paddingTop": 12,
            "width": "75%",
          },
          undefined,
        ]
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "fontSize": 14,
              "height": 21,
              "paddingHorizontal": 6,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#e24d4d",
          "fontSize": 11,
          "margin": 24,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#ccc",
            "borderRadius": 3,
            "borderWidth": 1.5,
            "marginVertical": 0,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "500",
              },
              undefined,
            ]
          }
        >
          Log In
        </Text>
      </View>
    </View>
  </View>
</View>
`;
